gems/quilt_rails/test/controllers/quilt/ui_controller_test.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "action_controller"
+
+module Quilt
+  class UiControllerTest < ActionDispatch::IntegrationTest
+    include ActiveSupport::Testing::Isolation
+
+    def run(*)
+      with_rails_env("development") { super }
+    end
+
+    setup { boot_dummy }
+
+    test "react render error" do
+      get("/")
+
+      assert_select("h1", "Waiting for React Sever to boot up.")
+      assert_select("meta[http-equiv='refresh']") do |selector, *|
+        assert_equal("3;URL='/'", selector[:content])
+      end
+    end
+
+    private
+
+    def with_rails_env(environment)
+      previous_environment = Rails.env.to_s
+      Rails.env = environment
```

Comment:
ENV['RAILS_ENV'] . In the forked process, Rails.env should have no values. This is also the normal way to boot an app with whatever environment (i.e. you don't do `Rails.env = 'test' bin/rails c`).

You also don't need to reset the ENV because when the test starts you are already in a forked process, the env will just "die" when the test ends (it won't bubble up to the main process)

Reply:
I understand that, but if a previous test run memoizes `Rails.env` the fork seems to inherit the value, and then the env var has no effect. At least that's what I'm seeing.